lib/elixir/lib/registry.ex

```diff
@@ -976,6 +976,54 @@ defmodule Registry do
     end
   end
 
+  @doc """
+  Counts registered keys.
+
+  ## Examples
+  In the example below we register the current process and count the number
+  of registered keys:
+
+      iex> Registry.start_link(keys: :unique, name: Registry.UniqueCountTest)
+      iex> Registry.count(Registry.UniqueCountTest)
+      0
+      iex> {:ok, _} = Registry.register(Registry.UniqueCountTest, "hello", :world)
+      iex> {:ok, _} = Registry.register(Registry.UniqueCountTest, "world", :world)
+      iex> Registry.count(Registry.UniqueCountTest)
+      2
+
+  The same applies to duplicate registries:
+
+      iex> Registry.start_link(keys: :duplicate, name: Registry.DuplicateCountTest)
+      iex> Registry.count(Registry.DuplicateCountTest)
+      0
+      iex> {:ok, _} = Registry.register(Registry.DuplicateCountTest, "hello", :world)
+      iex> {:ok, _} = Registry.register(Registry.DuplicateCountTest, "hello", :world)
+      iex> Registry.count(Registry.DuplicateCountTest)
+      2
+  """
+  @spec count(registry) :: integer()
+  def count(registry) when is_atom(registry) do
+    case key_info!(registry) do
+      {_kind, partitions, nil} ->
+        Enum.reduce(0..(partitions - 1), 0, fn partition_index, acc ->
+          [{^partition_index, partition_ets, _}] = :ets.lookup(registry, partition_index)
+          size = safe_count(partition_ets)
+          acc + size
+        end)
+
+      {_kind, 1, key_ets} ->
+        safe_count(key_ets)
+    end
+  end
+
+  defp safe_count(ets) do
+    try do
+      :ets.info(ets, :size)
```

Comment:
Thank you for the PR!

I could not find documentation about the performance of this operation. I am assuming it happens in constant time, so maybe we should call it `size`.

However, we may also want to introduce `count_match`, which then is definitely linear, so maybe sticking with the `count` vocabulary is the safest choice?

Thoughts @elixir-lang/elixir?

Reply:
Looking at the Erlang source it looks like it is a constant time operation

https://github.com/erlang/otp/blob/2960c7d08932b718dc93773e0718cd214277e09d/erts/emulator/beam/erl_db.c#L3999

I'm ok with renaming it to `size`